Woopra/Woopra-iOS
# Prepare for Swift Package Manager

### Purpose:
1. Prepare `Package.swift` for supporting Swift Package Manager.
2. Upgrade to use swift v5.0 as v4.0 is a really old version.
3. Ensure the name in the `podspec` matches the one in `Package.swift` to avoid confusion, i.e. using `import Woopra-iOS` in CocoaPods and `import WoopraSDK` in Swift Package Manager.
4. ~~Fix some errors coming from running `swift build`. The changes I made were to specifically add `import Foundation` and check if we can use `UIKit`. The errors include:~~
```
error: @objcMembers attribute used without importing module 'Foundation'
```
```
error: cannot find 'UIDevice' in scope
```
5. ~~I changed the directory structure by separating Swift and Objective-C tests, enabling inclusion of the `testTarget` and enabling the ability to run `swift test`.~~
6. Update the README.md

### Expected Behavior:
There should not be any behavior changes.

### How To Verify it:
~~1. Run `swift build` in the root directory without errors~~
~~2. Run `swift test` in the root directory without failed cases.~~
3. Run `Test` in Xcode without errors.
4. [Updates] As a member of SPM mentioned in https://github.com/apple/swift-package-manager/issues/6571#issuecomment-1551952258

> Note that building with SwiftPM for iOS etc will not be a valid way to verify your package entirely since it won't 1:1 match the behavior of Xcode's build system (especially around resources, but not limited to that). That said, I think it's still a reasonable enhancement to allow building for other Apple platforms using SwiftPM.

I suppose we can build the package in Xcode to verify it(by selecting the Scheme `WoopraSDK` and target `Any iOS Device(arm64)`) instead of running `swift build`.


### Note:
1. As We changed the name in `podspec` in this PR, we should make another publish to replace the current version with the new name.
2. We should update the tag.
